.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,44 @@
+name: Pull Request
+
+
+on:
+  pull_request:
+    branches:
+    - main
+
+
+env:
+  PYTHON: python
+
+
+jobs:
+  cancel_previous:
+    name: 'Cancel Previous Runs'
+    runs-on: ubuntu-20.04
+    timeout-minutes: 3
+
+    steps:
+    - uses: ydataai/cancel-workflow-action@0.7.0
+      with:
+        ignore_sha: true
+        access_token: ${{ secrets.ACCESS_TOKEN }}
+
+
+  validate:
+    name: Validate
+    runs-on: ubuntu-20.04
+
```

Comment:
```
needs:
- cancel_previous
```

Reply:
why? the idea is to cancel because it's not necessary to run it fully, but it can start to run right away

Reply:
also, in this case it doesn't really need to cancel, this will be really quick